doc/developers/contributing.rst

```diff
@@ -952,6 +952,18 @@ The easiest and recommended way to accomplish this is to
 All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
+Parameters can be passed to the ``fit`` method. Generally, these parameters
+should be restricted to variables that have the shape ``n_samples``.
```

Comment:
is `shape` the right word to use here / i think an example would be good for those unfamiliar with terminology? I feel like it might be clearer to say "variables with `n_samples` elements, such as the data `X` and the labels `y`"